modules/@apostrophecms/doc-type/index.js

```diff
@@ -520,11 +520,17 @@ module.exports = {
 
       async convert(req, input, doc, options = {
         presentFieldsOnly: false,
-        copyingId: false
+        copyingId: false,
+        isPage: false
       }) {
-        const fullSchema = self.apos.doc.getManager(options.type || self.name).allowedSchema(req);
+        let fullSchema;
         let schema;
         let copyOf;
+        if (options.isPage) {
```

Comment:
It would be better classical OOP practice if this were an invocation of a new doc-type method that is overridden for `page-type`, rather than "if this type else this type" logic.

In fact, probably `allowedSchema` should just take an `options` object to allow override of `type`.

Reply:
@boutell It seems like `allowedSchema` should be that method. The problem I'm running into is that the page module's `allowedSchema` calls the doc-type's `allowedSchema`, setting up an infinite loop.

What I think we should do is move the page module's `allowedSchema` functionality into `page-type`, using  `extendMethods` to use the original `doc-type` method (since the page version already does), but wrapped in the extra page functionality. There are three or four page module methods in there that we could simply use with `apos.page.whatever`. There's one place in the page module where we'd change `self.allowedSchema` to `manager.allowedSchema`.

Do you see any issues with that?

Reply:
I agree that's the best route, with a compatibility wrapper for anyone who does call `apos.page.allowedSchema`.